### DISCUSSION TITLE
Registration UI not working

### DISCUSSION THREAD

**Discussion by jaschaio** [plausible/analytics#4291] [Self-Hosted Support] [answered]
I have just started a fresh install from the community edition following the readme.

My `plausible-conf.env` contains values for all of these:

```
BASE_URL=
SECRET_KEY_BASE=
TOTP_VAULT_KEY=

GOOGLE_CLIENT_ID=
GOOGLE_CLIENT_SECRET=

DISABLE_REGISTRATION=false
```

I can see the registration page in the browser but when I click on "Create my account" nothing happens.

Using `v2.1.1`

**Comment by ruslandoga** [comment, marked as answer]
👋 @jaschaio 

Please check the browser console and `docker compose logs` for errors.

**Reply by jaschaio** [reply]
Thanks, after checking the console I saw that some errors where hidden because "Hide network" was activated.

<img width="758" alt="image" src="https://github.com/plausible/analytics/assets/11058666/871079ad-b24e-4909-91eb-ba5e67e36fe1">

After deactivating that console checkbox I noticed it was because of the same [websocket errors as here](https://github.com/plausible/analytics/discussions/3826).

This solution worked https://github.com/plausible/analytics/discussions/3826#discussioncomment-8582240

**Reply by ruslandoga** [reply]
Great! Note however that some users complained about higher memory usage with that config. [The one in the repo](https://github.com/plausible/community-edition/blob/e023cf8874871865be88d81b759c62d1032237a6/reverse-proxy/nginx/plausible#L12-L17) is supposed to work better somehow. More info about the change: https://github.com/plausible/community-edition/pull/126

**Reply by ballistix-i7** [reply]
> Great! Note however that some users complained about higher memory usage with that config. [The one in the repo](https://github.com/plausible/community-edition/blob/e023cf8874871865be88d81b759c62d1032237a6/reverse-proxy/nginx/plausible#L12-L17) is supposed to work better somehow. More info about the change: [plausible/community-edition#126](https://github.com/plausible/community-edition/pull/126)

Anyone here with directadmin and openlitespeed? I have the same problem, click on register and nothing happens. I don't know how to make these changes in openlitespeed conf :(

**Reply by ruslandoga** [reply]
👋 @ballistix-i7 

I don't think many people here use OpenLiteSpeed so it might make sense to ask that question elsewhere.

<details><summary>Here's what ChatGPT responded with when provided with the NGINX config as example</summary>

---

To adapt the Nginx configuration for **DirectAdmin with OpenLiteSpeed**, you'll need to:

1. Translate the Nginx configuration into OpenLiteSpeed's format.
2. Add specific directives for handling WebSockets.

Here’s how to approach the problem and the required changes:

---

### **Step 1: DirectAdmin and OpenLiteSpeed Basics**

- **DirectAdmin**: A web hosting control panel. When using OpenLiteSpeed with DirectAdmin, configurations are typically done through the DirectAdmin Custom HTTPD feature or directly in OpenLiteSpeed's web admin interface.
- **OpenLiteSpeed**: Uses XML-based configuration files or a web admin interface for settings. Unlike Nginx, OpenLiteSpeed requires creating listeners and context mappings explicitly.

---

### **Step 2: Convert Nginx Config to OpenLiteSpeed**

For the given Nginx config:

#### Nginx Config Equivalent in OpenLiteSpeed:
1. **Set Up a Listener**: 
   OpenLiteSpeed listens for traffic on port `80` by default. If not configured, add a listener in the OpenLiteSpeed Web Admin under **Listeners**.

2. **Virtual Host Configuration**:  
   In the OpenLiteSpeed Web Admin, navigate to your Virtual Host settings. Add a **Rewrite Rule** and a **Context** to handle WebSocket requests.

#### Example OpenLiteSpeed Configuration:
```xml
# OpenLiteSpeed Configuration for example.com

# Listener for HTTP traffic
listener Default {
    address *:80
    secure 0
}

# Virtual Host Config
virtualHost Example {
    vhRoot /path/to/your/site
    domain example.com
    enableGzip 1

    # Rewrite rules
    rewrite {
        enable 1
        rewriteFile /path/to/your/site/.htaccess
    }

    # Add a context for WebSocket support
    context /live/websocket {
        type proxy
        uri http://127.0.0.1:8000
        allowBrowse 1

        # Enable WebSocket headers
        addResponseHeader Upgrade $http_upgrade
        addResponseHeader Connection "Upgrade"
    }

    # Default context for other paths
    context / {
        type proxy
        uri http://127.0.0.1:8000
        allowBrowse 1
    }
}
```

---

### **Step 3: Enable WebSockets**

To ensure WebSockets work, you need to:
1. **Enable HTTP/1.1**: OpenLiteSpeed uses HTTP/1.1 by default for proxying, but confirm it explicitly in the proxy settings.
2. **Add Upgrade Headers**:
   - `Upgrade` and `Connection: Upgrade` are essential for WebSocket support.
   - These headers are added in the **Context** configuration for `/live/websocket`.

---

### **Step 4: Applying Changes via DirectAdmin**

In DirectAdmin, you can modify OpenLiteSpeed's configuration as follows:
1. **Access OpenLiteSpeed's Admin Panel**:
   - Log in at `http://your-server:7080` (default port for OpenLiteSpeed admin panel).
2. **Navigate to Virtual Hosts**:
   - Go to **Virtual Hosts** → Select your virtual host (e.g., `example.com`).
3. **Add Context** for `/live/websocket`:
   - Under **Context**, add a `Proxy` context.
   - Set `URI` to `http://127.0.0.1:8000`.
   - Add `Upgrade` and `Connection: Upgrade` headers under **Headers**.
4. **Modify Rewrite Rules**:
   - Add a rewrite rule to handle WebSocket routes if needed.

---

### **Testing the Configuration**

1. Restart OpenLiteSpeed after saving changes:
   ```bash
   service lsws restart
   ```
2. Test the `/live/websocket` route with tools like `curl` or browser dev tools to confirm WebSocket requests are being forwarded correctly.

---

</details>
